nengo/solvers.py

```diff
@@ -404,8 +408,10 @@ def __call__(self, A, Y, rng=None, E=None):
         import scipy.optimize
 
         tstart = time.time()
-        Y, m, n, d, matrix_in = format_system(A, Y)
-        Y = self.mul_encoders(Y, E)
+        Y, m, n, _, matrix_in = format_system(A, Y)
+        Y = self.mul_encoders(Y, E, copy=True)
+        d = Y.shape[1]
+        Y[Y < 0] = 0  # makes the final RMS error more reasonable
```

Comment:
Oh no, you're right, this is bad. I liked it because it helped to compare different solvers when I was just playing around, but you're right that it misrepresents things. It could also be problematic if we ever have neurons that can have negative activities.